src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -125,40 +125,32 @@ public virtual void EvaluatePreflightRequest(HttpContext context, CorsPolicy pol
             var requestHeaders =
                 context.Request.Headers.GetCommaSeparatedValues(CorsConstants.AccessControlRequestHeaders);
 
-            if (!policy.AllowAnyMethod)
+            if (!policy.AllowAnyMethod && !policy.Methods.Contains(accessControlRequestMethod.ToString(), StringComparer.OrdinalIgnoreCase))
             {
-                var found = false;
-                for (var i = 0; i < policy.Methods.Count; i++)
-                {
-                    var method = policy.Methods[i];
-                    if (string.Equals(method, accessControlRequestMethod, StringComparison.OrdinalIgnoreCase))
-                    {
-                        found = true;
-                        break;
-                    }
-                }
-
-                if (!found)
-                {
-                    _logger?.PolicyFailure();
-                    _logger?.AccessControlMethodNotAllowed(accessControlRequestMethod);
-                    return;
-                }
+                _logger?.PolicyFailure();
+                _logger?.AccessControlMethodNotAllowed(accessControlRequestMethod);
+                return;
```

Comment:
A for loop seemed to be an useless workaround used for filtering using ignorecase. I just changed it to whatever the rest was using as a reflex, didn't know I commited this for this fix. I'm happy to reverting it if you think it needs reverting.

Reply:
@FlyingWraptor We just don't want to change code that is not related to a given fix as it can have unexpected consequences that require further testing. (No matter how small or trivial it seems).

I hope that makes sense.

Reply:
Understood, will revert right now.